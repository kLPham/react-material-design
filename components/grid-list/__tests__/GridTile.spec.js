import React from 'react';
import { shallow } from 'enzyme';
import GridTile from '../gridTile';

const img = 'https://static.pexels.com/photos/58997/pexels-photo-58997.jpeg';

describe('<GridTile />', () => {
    it('should be able to render without a title', () => {
        const gridTile = shallow(
            <GridTile src={img} />,
    );
        expect(gridTile).toMatchSnapshot();
    });
    it('should be able to render a title', () => {
        const title = 'This is my title';
        const gridTile = shallow(
            <GridTile src={img} title={title} />,
      );
        expect(gridTile).toMatchSnapshot();
    });
    it('should be able to render an icon', () => {
        const title = 'This is my title';
        const icon = 'menu';
        const gridTile = shallow(
            <GridTile src={img} title={title} icon={icon} />,
    );
        expect(gridTile).toMatchSnapshot();
    });
    it('should be able to render support text', () => {
        const title = 'This is my title';
        const supportText = 'This is my support text.';
        const gridTile = shallow(
            <GridTile src={img} title={title} supportText={supportText} />,
    );
        expect(gridTile).toMatchSnapshot();
    });
    it('should be able to render icon and support text', () => {
        const title = 'This is my title';
        const icon = 'menu';

        const supportText = 'This is my support text.';
        const gridTile = shallow(
            <GridTile src={img} title={title} icon={icon} supportText={supportText} />,
    );
        expect(gridTile).toMatchSnapshot();
    });
});
