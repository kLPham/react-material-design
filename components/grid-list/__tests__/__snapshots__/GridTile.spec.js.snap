// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTile /> should be able to render a title 1`] = `
<li
  className="mdc-grid-tile"
>
  <div
    className="mdc-grid-tile__primary"
  >
    <div
      className="mdc-grid-tile__primary-content my-tile-image"
      style={
        Object {
          "backgroundImage": "url(https://static.pexels.com/photos/58997/pexels-photo-58997.jpeg)",
        }
      }
    />
  </div>
  <span
    className="mdc-grid-tile__secondary"
  >
    <span
      className="mdc-grid-tile__title"
    >
      This is my title
    </span>
  </span>
</li>
`;

exports[`<GridTile /> should be able to render an icon 1`] = `
<li
  className="mdc-grid-tile"
>
  <div
    className="mdc-grid-tile__primary"
  >
    <div
      className="mdc-grid-tile__primary-content my-tile-image"
      style={
        Object {
          "backgroundImage": "url(https://static.pexels.com/photos/58997/pexels-photo-58997.jpeg)",
        }
      }
    />
  </div>
  <span
    className="mdc-grid-tile__secondary"
  >
    <i
      className="mdc-grid-tile__icon material-icons"
    >
      menu
    </i>
    <span
      className="mdc-grid-tile__title"
    >
      This is my title
    </span>
  </span>
</li>
`;

exports[`<GridTile /> should be able to render icon and support text 1`] = `
<li
  className="mdc-grid-tile"
>
  <div
    className="mdc-grid-tile__primary"
  >
    <div
      className="mdc-grid-tile__primary-content my-tile-image"
      style={
        Object {
          "backgroundImage": "url(https://static.pexels.com/photos/58997/pexels-photo-58997.jpeg)",
        }
      }
    />
  </div>
  <span
    className="mdc-grid-tile__secondary"
  >
    <i
      className="mdc-grid-tile__icon material-icons"
    >
      menu
    </i>
    <span
      className="mdc-grid-tile__title"
    >
      This is my title
    </span>
    <span
      className="mdc-grid-tile__support-text"
    >
      This is my support text.
    </span>
  </span>
</li>
`;

exports[`<GridTile /> should be able to render support text 1`] = `
<li
  className="mdc-grid-tile"
>
  <div
    className="mdc-grid-tile__primary"
  >
    <div
      className="mdc-grid-tile__primary-content my-tile-image"
      style={
        Object {
          "backgroundImage": "url(https://static.pexels.com/photos/58997/pexels-photo-58997.jpeg)",
        }
      }
    />
  </div>
  <span
    className="mdc-grid-tile__secondary"
  >
    <span
      className="mdc-grid-tile__title"
    >
      This is my title
    </span>
    <span
      className="mdc-grid-tile__support-text"
    >
      This is my support text.
    </span>
  </span>
</li>
`;

exports[`<GridTile /> should be able to render without a title 1`] = `
<li
  className="mdc-grid-tile"
>
  <div
    className="mdc-grid-tile__primary"
  >
    <div
      className="mdc-grid-tile__primary-content my-tile-image"
      style={
        Object {
          "backgroundImage": "url(https://static.pexels.com/photos/58997/pexels-photo-58997.jpeg)",
        }
      }
    />
  </div>
</li>
`;
